File: bindings/php/configurator.php
<?php

define('configurator_command', "/usr/local/bin/configurator");

$schemas_file = null;
$configs_file = null;
$configurations = array();

class Configuration {
    var $name = '';
    var $options = array();
    
    function __constructor__($name) {
        $this->name = $name;
        $this->options = array();
    }

    function get($name) {
        return $this->options[$name];
    }

    function set($name, $value) {
        $this->options[$name] = $value;
    }
}

function configurator_setup($schemas, $configs) {
    global $schemas_file;
    global $configs_file;
    
    $schemas_file = $schemas;
    $configs_file = $configs;

    load_configurations();
}

function configurator_setup_example() {
    configurator_setup(__FILE__ . "../../doc/example/configurator.schema",
                       __FILE__ . "../../doc/example/configurator.config");
}

function configurator_list_configs() {
    global $schemas_file;
    global $configs_file;
    
    $output = exec(constant('configurator_command') . ' --schemas ' . $schemas_file .
                    ' --configs ' . $configs_file . ' --list --json');
    return json_decode($output, true);
}

function configurator_inspect($config) {
    global $schemas_file;
    global $configs_file;
    
    $output = exec(constant('configurator_command') . ' --schemas ' . $schemas_file .
                    ' --configs ' . $configs_file . ' -i ' . $config . ' --json');
    return json_decode($output, true);
}

function load_configurations() {
    global $configurations;

    $configurations = array();
    
    foreach (configurator_list_configs() as $config) {
        $configuration = new Configuration($config);

        foreach (configurator_inspect($config) as $option) {
            $configuration->set($option['option'], $option['value']);
        }

        $configurations[$config] = $configuration;        
    }
}

function get_configuration($name) {
    global $configurations;

    return $configurations[$name];

}

?>

File: bindings/python/configurator.py
import os
import subprocess
import json
import sys

configurator_command = "/usr/local/bin/configurator"

schemas_file = None
configs_file = None
configurations = {}

def configurator_setup(schemas, configs):
    global schemas_file
    global configs_file
    schemas_file = schemas
    configs_file = configs
    
    load_configurations()
    
def configurator_setup_example():
    configurator_setup(os.getcwd() + "/../../doc/example/configurator.schema",
                       os.getcwd() + "/../../doc/example/configurator.config")
    
class Configuration():
    def __init__(self, name):
        self._name = name
        self._options = {}
        
    def get(self, name):
        return self._options.get(name)
    
    def set(self, name, value):
        self._options[name] = value   
    
def load_configurations():
    global configurations
    
    configurations = {}
    
    for config in configurator_list_configs():
        data = configurator_inspect(config)
        configuration = Configuration(config)
        for option in data:
            configuration.set(option.get('option'),
                              parse_option_value(option))
        configurations[config] = configuration

def parse_option_value(option):
    return option.get('value')

def get_configuration(name):
    return configurations.get(name)

def configurator_list_configs():
    output = subprocess.check_output([configurator_command, '--schemas', schemas_file, '--configs', configs_file, '-list', '--json'])
    
    return json.loads(output)

def configurator_inspect(config):
    output = subprocess.check_output([configurator_command, '--schemas', schemas_file, '--configs', configs_file, '-i', config, '--json'])
    
    return json.loads(output)


if __name__ == '__main__':
    
    configurator_setup_example()
    
    configuration = get_configuration("Dev")
    
    print "Dev.Web server.Host=" + configuration.get("Web server.Host")